app/controllers/content_items_controller.rb

```diff
@@ -13,6 +13,10 @@ class ContentItemsController < ApplicationController
 
   attr_accessor :content_item, :taxonomy_navigation
 
+  content_security_policy do |p|
+    p.connect_src(*p.connect_src, -> { csp_connect_src })
```

Comment:
Just to check if I have understood correctly... in the past i've defined the CSP for the application in the initializers somewhere. In this case we want to change the CSP specific to different parts of the application (webchat), so this is instead setting the CSP for content items but it is overriden by this method in [contact_presenter](https://github.com/alphagov/government-frontend/blob/76071c744e4de0b5b5fed38de3758142d419fce5/app/presenters/contact_presenter.rb#L96) for webchat to allow the extra value? Otherwise `connect_src` is set to nil.

Reply:
Yup exactly that. 

We configure a global CSP for the app via [the initializer](https://github.com/alphagov/government-frontend/blob/1175a1a8d5ed3401bcf48dacd404ca7cf9753f7d/config/initializers/content_security_policy.rb#L1) and then we can refine it a controller or action level. 

This allows us to have three layers of CSP:
1) In govuk_app_config, directives that cover all GOV.UK apps. These are a pain to change and test because they require a new gem release, but are good to apply one rule consistently everywhere. Best used for a change that applies to multiple apps e.g. analytics or govspeak
2) Across an app, via the initializer. This offers us a chance to apply configuration that applies to just one app and every route in it. Much faster to change than the gem. Best used for a case where an app uses a library that other apps don't. There are no cases of this used yet, but I'm expecting it to be used so that apps without jQuery can be configured more securely than the apps that still use jQuery (licence-finder and Whitehall).
3) In a controller, which can be specified to an action level, this allows just a single resource to set up extra rules for the CSP which is best for when there's an exception to the rules for a couple of pages, i.e this scenario. We have precedence for doing this in Smart Answers already: https://github.com/alphagov/smart-answers/blob/45a463407b84ed3e58b268f4522d7493dbdd40ba/app/controllers/smart_answers_controller.rb#L8-L12

Reply:
ok cool, just wanted to check I wasn't missing something. Thanks for the explanation, makes sense :)